torch/_inductor/index_propagation.py

```diff
@@ -258,5 +258,13 @@ def indirect_indexing(
         if isinstance(index, IndexPropVar) and index.is_symbolic:
             # If we are turning a indirect indexing into direct, we need to wrap it.
             index = index.value.expr
+            # If SymPy can prove that idx >= 0, the Where below will evaluate to zero
+            # Alas, SymPy can't prove that index >= when we do ModularIndexing...
```

Comment:
I think you should put this in `ModularIndexing._eval_is_positive` so sympy will be able to infer this everywhere. Also this should probably be `nonnegative` rather than `positive` since `ModularIndexing` can give zero.

Reply:
ahhhh, that's the function I was looking for. I was just about to ping Aaron to ask him whether this could be generalised.